### PR TITLE
Add Hetzner docs

### PR DESCRIPTION
**Pull request by felixfontein**
##### SUMMARY
Add a guide to the Hetzner modules, including a migration guide.

The inventory plugin isn't documented yet; it's also missing in the HostTech guide. I'll work on that later.

##### ISSUE TYPE
- Docs Pull Request

##### COMPONENT NAME
Hetzner modules and plugins

### CODE REVIEW

**Comment by felixfontein** [comment]
A preview can be seen on https://ansible.fontein.de/collections/community/dns/docsite/hetzner_guide.html.

CC @markuman

File: docs/docsite/rst/hetzner_guide.rst [hunk 1]
```diff
@@ -0,0 +1,424 @@
...
+- :ref:`community.dns.hetzner_dns_record_set <ansible_collections.community.dns.hetzner_dns_record_set_module>`: create/update/delete DNS record sets
+- :ref:`community.dns.hetzner_dns_record_sets <ansible_collections.community.dns.hetzner_dns_record_sets_module>`: bulk synchronize DNS record sets
+- :ref:`community.dns.hetzner_dns_zone_info <ansible_collections.community.dns.hetzner_dns_zone_info_module>`: retrieve zone information
+
+If you are interested in migrating from the `markuman.hetzner_dns collection <https://galaxy.ansible.com/markuman/hetzner_dns>`_, please see :ref:`ansible_collections.community.dns.docsite.hetzner_guide.migration_markuman_hetzner_dns`.
+
+Authentication
+--------------
+
+To use Hetzner's API, you need to create a API token. You can manage API tokens in the "API tokens" menu entry in your user menu in the `DNS Console <https://dns.hetzner.com/>`_. You must provide the token to the ``hetzner_token`` option of the modules, its alias ``api_token``, or pass it on in the ``HETZNER_DNS_TOKEN`` environment variable:
```
Comment (markuman) [inline docs/docsite/rst/hetzner_guide.rst:26]:
```suggestion
To use Hetzner's API, you need to create an API token. You can manage API tokens in the "API tokens" menu entry in your user menu in the `DNS Console <https://dns.hetzner.com/>`_. You must provide the token to the ``hetzner_token`` option of the modules, its alias ``api_token``, or pass it on in the ``HETZNER_DNS_TOKEN`` environment variable:
```

**Review by markuman — APPROVED** [review]
LGTM so far

**Comment by felixfontein** [comment]
@markuman thanks a lot for reviewing this!

**Pull request merged**